src/core/queue.ts

```diff
@@ -481,36 +482,26 @@ export class Queue extends EventEmitter {
           failed_at: new Date().toString(),
         };
       }
+    }
 
-      await this.connection.redis.incr(this.connection.key("stat", "failed"));
-      await this.connection.redis.incr(
-        this.connection.key("stat", "failed", workerName)
+    const pipeline = this.connection.redis
+      .multi()
+      .incr(this.connection.key("stat", "failed"))
+      .del(this.connection.key("stat", "failed", workerName))
+      .del(this.connection.key("stat", "processed", workerName))
+      .del(this.connection.key("worker", "ping", workerName))
+      .del(this.connection.key("worker", workerName, queues, "started"))
+      .del(this.connection.key("worker", workerName))
+      .srem(this.connection.key("workers"), workerName + ":" + queues);
```

Comment:
This is the main enhancement - This method will now either remove all or none of the worker's data. no more partial deletes.